application.go

```diff
@@ -109,6 +129,28 @@ type DeleteAppOpts struct {
 	Force bool `url:"force,omitempty"`
 }
 
+// Diego ======
+
+// SetIPAddressPerTask defines that the application will have a IP address defines by a external agent. this
+// configuration is not allowed to be used with Port or PortDefinitions.
+func (r *Application) SetIPAddressPerTask(ipAddressPerTask *IPAddressPerTask) *Application {
+
+	r.Ports = make([]int, 0)
```

Comment:
Personally, I usually prefer to let the API do the error handling as it means less coupling. In this very case though, I think it's a convenience add-on we can justify to keep in. (Especially since the documentation is a bit incomplete here.)